source/mir/ndslice/slice.d

```diff
@@ -1907,9 +1932,15 @@ public:
             if (this._iterator == rslice._iterator)
                 return true;
         }
+
         import mir.algorithm.iteration : equal;
         static if (__traits(compiles, this.lightScope))
-            return equal(this.lightScope, rslice.lightScope);
+        {
+            foreach(i; Iota!(min(this.L, rslice.L)))
+                if(this.label!i != rslice.label!i)
+                    return false;
+            return equal(this.lightScope.values, rslice.lightScope.values);           
+        }
```

Comment:
Needs some optimisation.
First, define two vars for like`auto lhs = this.lightScope `.
And then use them both for labels and values.

Reply:
Addressed in 08d86b0